tests/textbox/api.js

```diff
@@ -505,4 +505,25 @@
 
         equal(textbox._text.attr("title"), 20);
     });
+
+    asyncTest("input blinks when invalid input is entered.", function () {
```

Comment:
The test can be split in 3 new tests, which test for:
1. The input doesn't have an invalid decoration (the first 2 assertions).
2. The keypress event adds invalid decoration (the next 2 assertions).
3. A keypress event is triggered and after a certain time, the invalid decoration is removed (the last 2 assertions).

You can also think about adding a new test file, e.g. `validation.js`, which seems to be more appropriate for such tests in comparison to `api.js`.